File: practice-projects/springboot4ddd/src/main/java/com/github/microwind/springboot4ddd/infrastructure/repository/order/OrderDO.java
package com.github.microwind.springboot4ddd.infrastructure.repository.order;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单数据对象（Data Object）
 *
 * <p>仅承担"与 orders 表字段一对一映射"的职责，所有 Spring Data JDBC / MyBatis-Plus
 * 注解都集中在这里。领域模型 {@code Order} 不再背负持久化细节。
 *
 * <p>转换由 {@link OrderConverter} 显式完成。
 *
 * @author jarry
 * @since 1.0.0
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("orders")
@TableName("orders")
public class OrderDO {

    @Id
    @TableId
    private Long id;

    @Column("order_no")
    private String orderNo;

    @Column("user_id")
    private Long userId;

    @Column("total_amount")
    private BigDecimal totalAmount;

    private String status;

    @Column("created_at")
    private LocalDateTime createdAt;

    @Column("updated_at")
    private LocalDateTime updatedAt;
}

File: practice-projects/gin-ddd/internal/infrastructure/persistence/order/order_do.go
// Package order 提供订单聚合根的持久化适配。
//
// OrderDO 承担"与 orders 表字段一对一映射"的职责,所有 db tag 仅集中于此,
// 领域模型 *order.Order 不再背负持久化细节。
package order

import "time"

// OrderDO 订单数据对象。
type OrderDO struct {
	ID          int64     `db:"id"`
	OrderNo     string    `db:"order_no"`
	UserID      int64     `db:"user_id"`
	TotalAmount float64   `db:"total_amount"`
	Status      string    `db:"status"`
	CreatedAt   time.Time `db:"created_at"`
	UpdatedAt   time.Time `db:"updated_at"`
}
